torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -612,38 +603,9 @@ def _sharded_pre_load_state_dict_hook(
             tensor, local_tensor, group=fsdp_state.process_group
         )
         tensor = tensor.narrow(0, 0, param_numel).reshape(param.size())
-        tensors_to_flatten.append(tensor)
+        state_dict[fqn_from_global_root] = tensor
 
-    # Create a new flat_param from the loaded, non-sharded tensors.
```

Comment:
I am guessing all of this logic got factored into the `_enter_unshard_params_ctx()`?

Reply:
Yes, the logic now is similar to full state_dict. So the main logic is to use `_enter_unshard_params_ctx()`.